adobe/spectrum-web-components
# feat: wrap hover rules in a media query

## Description
Expand the processing of Spectrum CSS output to wrap al usage of `:hover` in `@media (hover: hover) {}`.

In mobile delivery the `:hover` state is synthesized as a "sort of" first touch of an element (which causes the need to tab twice in some instances), and is difficult to remove, without passing the state to another element, which can cause the state to hangout passed it's usefulness. Wrapping the `:hover` rule in the media query prevents it from applying in touch only contexts. However, in some cases "active" styling and "hover" styling as _possibly_ conflated in a way that we'd need the Spectrum CSS source to take into account that we're doing this.

This effects the delivery of the following patterns:
- Accordion
- Action Button
- Button
- Card
- Checkbox
- Clear Button
- Close Button
- Dropzone
- Link
- Menu Item
- Number Field
- Picker Button
- Picker
- Radio
- Search
- Side Nav Item
- Slider
- Split View
- Switch
- Table Head Cell
- Table Row
- Tab
- Tag
- Textfield

Before moving forward, we'd need to review these patterns with design in a touch context to ensure we're continuing to deliver the experience specified for that environment.

## Related issue(s)
- fixes #2598

## How has this been tested?
    1. Visit each of the patterns above in [Storybook](https://control-hover--spectrum-web-components.netlify.app/storybook/) in a touch context
    2. Ensure that when touching them, then surface the expected "active" states.
    1. Visit each of the patterns above in [Storybook](https://control-hover--spectrum-web-components.netlify.app/storybook/) in a hover context, mouse/touch pag
    2. Ensure that when _hovering_ them, then surface the expected "hover" states.

## Types of changes

## Checklist
